File: backend/components/metiers/administration_publique/templates/helpers/core/templates_helper.test.py
# templates_helper.test.py – Tests unitaires Python pour helpers templates Threed
import unittest
import pytest
from backend.components.metiers.threed.templates.helpers.templates_helper import render_template, is_valid_template, mock_template_context, TemplatesHelper

class TestTemplatesHelper(unittest.TestCase):
    def test_render_template(self):
        self.assertTrue(callable(render_template))
    def test_is_valid_template(self):
        self.assertTrue(is_valid_template('rapport_audit.html.j2'))
        self.assertTrue(is_valid_template('foo.txt'))
        self.assertFalse(is_valid_template('foo.exe'))
    def test_mock_template_context(self):
        self.assertIsInstance(mock_template_context(), dict)

def test_templates_helper_init():
    helper = TemplatesHelper(options={"mode": "test"})
    assert helper.options == {"mode": "test"}
    assert helper.get_audit_trail() == []

def test_templates_helper_config_and_audit():
    helper = TemplatesHelper()
    assert helper.init({"version": 1}) is True
    assert helper.config == {"version": 1}
    assert helper.get_audit_trail()[0]["action"] == 'init'

def test_templates_helper_assist_valid():
    helper = TemplatesHelper()
    helper.init({"version": 2})
    result = helper.assist("OP", {"foo": "bar"})
    assert result == {"success": True, "operation": "OP", "data": {"foo": "bar"}, "config": {"version": 2}}
    assert len(helper.get_audit_trail()) == 2
    assert helper.get_audit_trail()[1]["action"] == 'assist'

def test_templates_helper_assist_invalid():
    helper = TemplatesHelper()
    with pytest.raises(ValueError):
        helper.assist(None, {})
    assert helper.get_audit_trail()[0]["action"] == 'error'

if __name__ == '__main__':
    unittest.main()

File: backend/components/metiers/administration_publique/templates/helpers/core/templates_helper.test.js
// templates_helper.test.js – Tests unitaires JS pour helpers templates Threed
const helper = require('./templates_helper');
describe('Helpers templates Threed (JS)', () => {
  test('renderTemplate fonctionne', () => {
    expect(typeof helper.renderTemplate).toBe('function');
  });
  test('isValidTemplate fonctionne', () => {
    expect(helper.isValidTemplate('rapport_audit.html.j2')).toBeTruthy();
    expect(helper.isValidTemplate('foo.txt')).toBeTruthy();
    expect(helper.isValidTemplate('foo.exe')).toBeFalsy();
  });
  test('mockTemplateContext retourne un objet', () => {
    expect(typeof helper.mockTemplateContext()).toBe('object');
  });
});

const TemplatesHelper = require('./templates_helper');

describe('TemplatesHelper', () => {
  it('doit s\'initialiser correctement', () => {
    const helper = new TemplatesHelper({ mode: 'test' });
    expect(helper.options).toEqual({ mode: 'test' });
    expect(helper.getAuditTrail()).toEqual([]);
  });

  it('doit initialiser la config et auditer', () => {
    const helper = new TemplatesHelper();
    expect(helper.init({ version: 1 })).toBe(true);
    expect(helper.config).toEqual({ version: 1 });
    expect(helper.getAuditTrail()[0].action).toBe('init');
  });

  it('doit assister une opération valide et auditer', () => {
    const helper = new TemplatesHelper();
    helper.init({ version: 2 });
    const result = helper.assist('OP', { foo: 'bar' });
    expect(result).toEqual({ success: true, operation: 'OP', data: { foo: 'bar' }, config: { version: 2 } });
    expect(helper.getAuditTrail().length).toBe(2);
    expect(helper.getAuditTrail()[1].action).toBe('assist');
  });

  it('doit lever une erreur sur une opération invalide et auditer', () => {
    const helper = new TemplatesHelper();
    expect(() => helper.assist(null, {})).toThrow('Invalid operation');
    expect(helper.getAuditTrail()[0].action).toBe('error');
  });
});
